rfc/rfc-43/rfc-43.md

```diff
@@ -0,0 +1,222 @@
+
+The Meta Table has the following features:
+
+1. Is a HUDI MOR Table
+
+2. Defines its own custom payload HoodieServicePayload
+
+3. Use basePath + instant as recordKey to guarantee unique
+
+4. Follows all the conventions of a HUDI Table
+
+### Scheduler
+
+Periodically scan the Meta Table and submit compaction/clustering job according to user-specified rules, and need to plug-in the execution engine.
+
+### Schema
+
+Being an actual Hoodie Table, the Meta Table needs to have a schema to store the compaction/clustering message. 
+
+```
+{
+    "namespace":"org.apache.hudi.avro.model",
+    "type":"record",
+    "name":"HoodieServiceRecord",
+    "doc":"Compaction/Clustering message saved within the Meta Table",
+    "fields":[
+        {
+            "name":"table_name",
+            "type":"string"
+        },
+        {
+            "name":"base_path",
+            "type":"string"
+        },
+        {
+            "name":"instant",
+            "doc":"compaction/clustering instant for the hoodie table",
+            "type":"string"
+        }, {
+            "name":"action",
+            "doc":"compact or cluster for this instant",
+            "type":"string"
+        },{
+            "name":"status",
+            "doc":"Status of the compaction/clustering job, 0：Scheduled 1：Running 2：Retrying 3:  Completed",
+            "type":"int"
+        } ,{
+            "name":"isDeleted",
+            "doc":"True if this plan has been deleted",
+            "type":"int"
+        }
+    ]
+}
+```
+
+## Interface design
+
+### Register
+
+/v1/hoodie/service/register
+
+```
+{
+"table_name":"table_name",
+"base_path":"/hoodie/base_path",
+"compaction_config":"compaction_config",
+"clustering_config":"clustering_config"
+}
+```
+
+### Schedule
+
+/v1/hoodie/service/schedule
+
+```
+{
+"table_name":"table_name",
+"base_path":"/hoodie/base_path",
+"action":"compact/cluster",
+"instant":"202111161559"
+}
+```
+
+
+### Delete
+
+/v1/hoodie/service/delete
+
+```
+{
+"table_name":"table_name",
+"base_path":"/hoodie/base_path",
+"action":"compact/cluster",
+"instant":"202111161559"
+}
+```
+
+## Error Handling
+
+Due to two tables being involved in each operation, we need to ensure that they are in sync and errors during dataset operation are handled correctly.
+
+1. Client scheduled the plan but request failed.
+
+2. RequestHandler received request but the commit is not completed.
+
+3. Client rollback plan after request to Compaction/Clustering Service.
+
+Client will report to service all unCompleted plans by time or num deltaCommits or combination, the service upsert to meta table uses HoodieServicePayload to guarantee any plan unique. Client use Delete request to rollback plan for Compaction/Clustering Service.
+
+### Configuration
+
+hoodie.service.enable=true
+
+hoodie.service.execute.parallelism=1
+
+## Proposal
+
+### Proposal For V1
+
```

Comment:
Maybe have a python / GO CLI too for the following:
1. List all jobs
2. Add job (useful for testing as well as manually adding operations for tables)
3. Remove job
4. Clear jobs for table (etc)

Reply:
Can we put the implementation of the client in the next stage, I hope the goal of the first step is to be able to complete the task execution of compaction & clustering job.

Reply:
@yuzhaojing please go ahead and incorporate/list out these future items in the RFC accordingly. then you may resolve the comment